matchers/vue/comments.go

```diff
@@ -0,0 +1,138 @@
+package vue
+
+import (
+	"github.com/preslavmihaylov/todocheck/matchers/state"
+)
+
+// NewCommentMatcher for vue comments
+func NewCommentMatcher(callback state.CommentCallback) *CommentMatcher {
+	return &CommentMatcher{
+		callback: callback,
+	}
+}
+
+// CommentMatcher for vue comments
+type CommentMatcher struct {
+	callback                  state.CommentCallback
+	buffer                    string
+	lines                     []string
+	linecnt                   int
+	stringToken               rune
+	isExitingMultilineComment bool
+	commentType               string
+	isStartingHTML            bool
+}
+
+// NonCommentState for vue comments
+func (m *CommentMatcher) NonCommentState(
+	filename, line string, linecnt int, prevToken, currToken, nextToken rune,
+) (state.CommentState, error) {
+	if currToken == '/' && prevToken == '/' {
+		m.buffer += string(currToken)
+
+		return state.SingleLineComment, nil
+	} else if currToken == '"' || currToken == '\'' {
```

Comment:
How is `\` a transition to a string state?

Reply:
It is not, but in this case it functions as a escape sequence to the ' symbol, so the go interpreter know that the literal '-rune is meant and the '-symbol is not the end of the rune sequence. I hope this makes sense to you, and as the new string-tests show, the transition seems to work. I therefore did not change this part of the code.

Reply:
Oh...

I didn't see that it's a `\'` instead of a `\` only. My bad :)